cmd/mesh/testdata/manifest-generate/input/ingressgateway_k8s_settings.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: install.istio.io/v1alpha2
+kind: IstioControlPlane
+spec:
+  hub: docker.io/istio
+  tag: 1.1.4
+  defaultNamespace: istio-control
+  trafficManagement:
+    enabled: false
+  policy:
+    enabled: false
+  security:
+    enabled: false
+  configManagement:
+    enabled: false
+  autoInjection:
+    enabled: false
+  gateways:
+    enabled: true
+    components:
+      ingressGateway:
+        enabled: true
+        k8s:
+          service:
+            externalTrafficPolicy: Local
```

Comment:
Some more complex example would be good. For example, adding to list of ports, which I'm actually not sure is going to work with this change...

Reply:
@ostromart I added the mirror of k8s service type so that unmarshalling can work with serviceport. PTAL

Reply:
lgtm, but could you hold off merging till https://github.com/istio/operator/pull/713 goes in? it will be easier to address any merge changes here rather than there.